paws_collector.js

```diff
@@ -495,6 +487,37 @@ class PawsCollector extends AlAwsCollector {
         });
     }
 
+    batchLogProcess(logs, privCollectorState, nextInvocationTimeout, callback) {
+        let collector = this;
+        return collector.processLog(logs, collector.pawsFormatLog.bind(collector), null,  (err) => {
+           
+            if (err) {
+                /**
+                 * MS api return the logs but if build payload size > 10MB, and split logs array the in two part.
+                 * Call the processLog again to send Logmsgs to ingest. 
+                 */
+                if ((typeof err === 'string' && err.indexOf('Maximum payload size exceeded') !== -1)) {
+
+                    let half = logs.length / 2;
+                    const indexArray = [
+                        { start: 0, stop: half },
+                        { start: half, stop: logs.length }
+                    ];
+
+                    async.reduce(indexArray, 0, (memo, logpart, asyncCallback) => {
```

Comment:
reduce returns a promise if 4th param `callback` is not pass
We need to add a new callback here and check the output of each `batchLogProcess`

Reply:
I think we can also use `map()` here, but I don't have strong preference.